### ISSUE TITLE
Lucene errors with latest opengrok source (Did you forget to add lucene-backward-codecs.jar?)

### ISSUE THREAD

**Issue by ebukoski** [OpenGrok/OpenGrok#1042]
I upgraded to the latest OpenGrok built directly from source.  Previously I was running that latest release version (0.12.1.5).  When I point the latest OpenGrok to the old index i get the stack trace below.

Is there a way to migrate the index after an upgrade or is a complete re-index the only solution here?  A complete reindex takes days for us.  BTW I tried adding  lucene-backward-codecs.jar but it didn't help.

```
java.lang.IllegalArgumentException: Could not load codec 'Lucene46'.  Did you forget to add lucene-backward-codecs.jar?
	at org.apache.lucene.index.SegmentInfos.readCodec(SegmentInfos.java:481)
	at org.apache.lucene.index.SegmentInfos.readCommit(SegmentInfos.java:361)
	at org.apache.lucene.index.StandardDirectoryReader$1.doBody(StandardDirectoryReader.java:53)
	at org.apache.lucene.index.StandardDirectoryReader$1.doBody(StandardDirectoryReader.java:50)
	at org.apache.lucene.index.SegmentInfos$FindSegmentsFile.run(SegmentInfos.java:731)
	at org.apache.lucene.index.StandardDirectoryReader.open(StandardDirectoryReader.java:50)
	at org.apache.lucene.index.DirectoryReader.open(DirectoryReader.java:63)
	at org.opensolaris.opengrok.web.SearchHelper.prepareExec(SearchHelper.java:261)
	at org.apache.jsp.search_jsp._jspService(search_jsp.java:158)
	at org.apache.jasper.runtime.HttpJspBase.service(HttpJspBase.java:70)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:723)
	at org.apache.jasper.servlet.JspServletWrapper.service(JspServletWrapper.java:388)
	at org.apache.jasper.servlet.JspServlet.serviceJspFile(JspServlet.java:313)
	at org.apache.jasper.servlet.JspServlet.service(JspServlet.java:260)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:723)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:290)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:233)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:191)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:127)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:103)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:109)
	at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:615)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:293)
	at org.apache.coyote.http11.Http11Processor.process(Http11Processor.java:861)
	at org.apache.coyote.http11.Http11Protocol$Http11ConnectionHandler.process(Http11Protocol.java:620)
	at org.apache.tomcat.util.net.JIoEndpoint$Worker.run(JIoEndpoint.java:489)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.IllegalArgumentException: An SPI class of type org.apache.lucene.codecs.Codec with name 'Lucene46' does not exist.  You need to add the corresponding JAR file supporting this SPI to your classpath.  The current classpath supports the following names: [Lucene54]
	at org.apache.lucene.util.NamedSPILoader.lookup(NamedSPILoader.java:109)
	at org.apache.lucene.codecs.Codec.forName(Codec.java:113)
	at org.apache.lucene.index.SegmentInfos.readCodec(SegmentInfos.java:469)
	... 27 more
```

**Comment by tarzanek** [comment]
We don't support upgrade. So new index please.

**Comment by ebukoski** [comment]
Ok thanks for confirming.  This is quite a pain for large codebases.

**Comment by tarzanek** [comment]
I know it's a PITA and that's why we're still not 1.x ;)
There is no upgrade code for indexes!
(it would be easy for lucene, but xrefs are not versioned, nor is history cache, so until this is fixed there will be no 1.0 ... )
